distributed/comm/tcp.py

```diff
@@ -436,9 +436,17 @@ async def connect(self, address, deserialize=True, **connection_args):
         kwargs = self._get_connect_args(**connection_args)
 
         try:
-            stream = await self.client.connect(
-                ip, port, max_buffer_size=MAX_BUFFER_SIZE, **kwargs
-            )
+            # server_hostname option (for SNI) only works with tornado.iostream.IOStream
```

Comment:
There have been discussions lately around removing `tornado` from distributed in favour of vanilla `asyncio`. How will that affect things here?

cc @fjetter

Reply:
Ripping out tornado entirely is still a bit away but I wouldn't want to introduce any new features that are tornado specific.

I barely know what this topic is about but reading tornado and asyncio docs shows that tornado is also just using  stdlib [`wrap_socket`](https://docs.python.org/3.7/library/ssl.html#ssl.SSLContext.wrap_socket) to implement this so I assume it would not be a big deal to add this to the `asyncio_tcp` backend.

@jcrist can you have a quick look?

Reply:
I maybe should have phrased this comment better.   `tornado.TCPClient.connect` doesn't take `server_hostname` as an argument; in order to establish a TLS connection with SNI with tornado, you need to first connect and then call `start_tls`.    

Glad you asked @jcrist to take a look, since I used the approach from dask-gateway: 
https://github.com/dask/dask-gateway/blob/main/dask-gateway/dask_gateway/comm.py#L45-L49
(based on advice from tornado devs https://github.com/tornadoweb/tornado/pull/2585#issuecomment-462080508)

Reply:
It looks like this can always be supported by any comm backend that uses SSLContext. If we have to we can hack it by providing an object with `wrap_socket` and `wrap_bio` methods that pass on the exact options we need.